Dockerfile.template

```diff
@@ -17,32 +17,26 @@ RUN  mkdir -p /opt/lsst/software/rspstack
 #  increases container size by about 60%.  If you want it in the same
 #  environment (the default), link the RSP loadstack instead.
 #COPY loadrspstack.bash ${LOADRSPSTACK}
+# In general: if an environment variable needs to be used across multiple
+#  stages, make it an ARG.  Introduce it before the first stage that uses it.
 RUN  ln -s ${LOADSTACK} ${LOADRSPSTACK}
 ARG  srcdir=/opt/lsst/src
 ARG  BLD=${srcdir}/build
-RUN  mkdir -p ${BLD}
+ARG  jl=/opt/lsst/software/jupyterlab
+ARG  verdir="${jl}/versions.installed"
```

Comment:
I don't understand why these are arguments. When would you ever want to change them? (The same for the pre-existing `ARG` lines above this.)

Reply:
It's not that you want to change them, exactly.  It's that you want to use them both in the main Dockerfile and inside the various shell stages.  ARG turns into an environment variable within the build context only (ENV stays an environment variable in the container runtime context).  So we want those around for concision during the build, but we don't want to persist them into the runtime environment.